planner/core/task.go

```diff
@@ -1591,28 +1591,29 @@ func (p *PhysicalHashAgg) attach2TaskForMpp(tasks ...task) task {
 		mpp.addCost(p.GetCost(inputRows, false))
 		return mpp
 	case Mpp2Phase:
-		// 2-phase agg: partial + final agg for hash partition
-		if len(p.MppPartitionCols) == 0 {
-			return invalidTask
-		}
-		prop := &property.PhysicalProperty{TaskTp: property.MppTaskType, ExpectedCnt: math.MaxFloat64, PartitionTp: property.HashType, PartitionCols: p.MppPartitionCols}
-		// if mpp does not need to enforce exchange, i.e., the child is properly partitioned, then this 2-phase agg is invalid
-		if !mpp.needEnforce(prop) {
-			return invalidTask
-		}
```

Comment:
why delete this case?

Reply:
I think 2 phase without a shuffle has no difference with 1 phase, right? Is there any test case that goes through this logic?

Reply:
I think 2 phase without a shuffle has no difference with 1 phase
->
no. if aggregations with 1 phase runs with 2 phase without a shuffle, this means these aggregations do twice. Even though the results are not wrong, increasing execution time, the plan does not right.

Test
->
properly yes, I do not remember.

Reply:
about the test
---
the related tests are passed properly because the cost is changed. If the using original cost, i.e., inputrows/2, some tests will not pass.
at this PR, 1-phase plan and 2-phase plan are generated in order and with the same cost, then the 1-phase is chosen but  the 2-phase is not rejected. if the plans are not generated in order, or not choose the first one from equal-cost plans, or the 2-phase plan is wrongly set to be lower than the 1-phase one, the cases are different.